packages/easy-testing/src/DataProvider/StaticFixtureFinder.php

```diff
@@ -18,17 +19,51 @@ final class StaticFixtureFinder
 {
     public static function yieldDirectory(string $directory, string $suffix = '*.php.inc'): Iterator
     {
-        $fileInfos = self::findFilesInDirectory($directory, $suffix);
+        return self::yieldFileInfos(self::findFilesInDirectory($directory, $suffix));
+    }
+
+    public static function yieldDirectoryExclusively(string $directory, string $suffix = '*.php.inc'): Iterator
+    {
+        return self::yieldFileInfosWithRelativePathname(self::findFilesInDirectoryExclusively($directory, $suffix));
+    }
+
+    public static function yieldDirectoryWithRelativePathname(string $directory, string $suffix = '*.php.inc'): Iterator
+    {
+        return self::yieldFileInfosWithRelativePathname(self::findFilesInDirectory($directory, $suffix));
+    }
+
+    public static function yieldDirectoryExclusivelyWithRelativePathname(
+        string $directory,
+        string $suffix = '*.php.inc'
+    ): Iterator {
+        return self::yieldFileInfosWithRelativePathname(self::findFilesInDirectoryExclusively($directory, $suffix));
+    }
+
+    /**
+     * @param SplFileInfo[] $fileInfos
+     */
+    private static function yieldFileInfos(array $fileInfos): Iterator
+    {
         foreach ($fileInfos as $fileInfo) {
-            yield [new SmartFileInfo($fileInfo->getRealPath())];
+            try {
+                $smartFileInfo = new SmartFileInfo($fileInfo->getRealPath());
+                yield [$smartFileInfo];
+            } catch (FileNotFoundException $e) {
+            }
         }
     }
 
-    public static function yieldDirectoryExclusively(string $directory, string $suffix = '*.php.inc'): Iterator
+    /**
+     * @param SplFileInfo[] $fileInfos
```

Comment:
We should have a return type to state `string` key. It's kind of main point of this method, so it should not be missed if this method is used.

Does PHPStan support something like this? 

```suggestion
     * @param SplFileInfo[] $fileInfos
     * @return Iterator<string, SplFileInfo>
```